src/protocol/bounce.rs

```diff
@@ -0,0 +1,116 @@
+use std::time::SystemTime;
+
+use serde::{de::Error, ser::*, Deserialize, Deserializer, Serialize, Serializer};
+use serde_json;
+use serde_json::Value;
+use serde_with::{serde_as, TimestampSeconds};
+
+#[derive(Debug, Clone)]
+pub struct Bounced {
+    pub games: Option<Vec<String>>,
+    pub slots: Option<Vec<i64>>,
+    pub tags: Vec<String>,
+    pub data: BounceData,
+}
+
+/// An internal representation of the [Bounced] struct, used as an intermediate
+/// state to determine how to decode the [BounceData].
+#[derive(Debug, Clone, Deserialize, Serialize)]
+struct InternalBounced {
+    pub games: Option<Vec<String>>,
+    pub slots: Option<Vec<i64>>,
+    #[serde(default)]
+    pub tags: Vec<String>,
+    pub data: Value,
+}
+
+// Deserialize Bounced based on its tags.
+impl<'de> Deserialize<'de> for Bounced {
+    fn deserialize<D>(deserializer: D) -> Result<Bounced, D::Error>
+    where
+        D: Deserializer<'de>,
+    {
+        let internal = InternalBounced::deserialize(deserializer)?;
+        if internal.tags.iter().any(|t| t == "DeathLink") {
+            Ok(Bounced {
+                games: internal.games,
+                slots: internal.slots,
+                tags: internal.tags,
+                data: BounceData::DeathLink(match serde_json::from_value(internal.data) {
+                    Ok(data) => data,
+                    Err(err) => return Err(D::Error::custom(err)),
+                }),
+            })
+        } else {
+            Ok(Bounced {
+                games: internal.games,
+                slots: internal.slots,
+                tags: internal.tags,
+                data: BounceData::Generic(internal.data),
+            })
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum BounceData {
+    DeathLink(DeathLink),
+    Generic(Value),
+}
+
+#[serde_as]
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct DeathLink {
+    #[serde_as(as = "TimestampSeconds<i64>")]
```

Comment:
```suggestion
    #[serde_as(as = "TimestampSeconds<f64>")]
```

Doc's say this one is a float and it wasn't deserializing properly until I changed it to f64